src/main/java/org/deepsymmetry/beatlink/data/OpusProvider.java

```diff
@@ -114,6 +114,14 @@ public FileSystem getFileSystem() {
      */
     private final Map<Integer, RekordboxUsbArchive> usbArchiveMap = new ConcurrentHashMap<>();
 
+    /**
+     * Contains a queue per slot number which allows us to slow down sending VirtualCdj.deliverMediaDetailsUpdate
+     * when we attach metadata archives until the application is actually ready to do so.
+     *
+     * Queues are initiated in constructor and should never be null.
+     */
+    private final Map<Integer, LinkedBlockingQueue<MediaDetails>> archiveAttachStatusMap = new ConcurrentHashMap<>();
```

Comment:
Queues were a great idea. Made it easy to find the right spot we have to wait for.